File: conference_speakers/src/main/java/com/microservices/conference_speakers/services/SpeakerService.java
package com.microservices.conference_speakers.services;

import com.microservices.conference_speakers.models.Speaker;

import java.util.List;

public interface SpeakerService {
    Speaker create(Speaker speaker);
    Speaker read(Long id);
    Speaker update(Long id, Speaker speaker);
    void delete(Long id);
    Speaker patch(Long id, Speaker speaker);
    List<Speaker> listSpeakers();
    String[] getNullPropertyNames(Speaker source, String... args);
    void deleteAll();
}

File: Angular10Crud/src/app/services/speaker.service.ts
import { Injectable } from '@angular/core';
import {HttpClient} from "@angular/common/http";
import {Observable} from "rxjs";
import { Speaker } from "../models/speaker.model";

const baseUrl = 'http://localhost:8000/api/version1/speakers';

@Injectable({
  providedIn: 'root'
})
export class SpeakerService {

  constructor(private http: HttpClient) { }

  getAll(): Observable<Speaker[]> {
    return this.http.get<Speaker[]>(baseUrl);
  }

  get(id: any): Observable<Speaker> {
    return this.http.get(`${baseUrl}/${id}`);
  }

  create(data: any): Observable<any> {
    return this.http.post(baseUrl, data);
  }

  update(id: any, data: any): Observable<any> {
    return this.http.put(`${baseUrl}/${id}`, data);
  }

  delete(id: any): Observable<any> {
    return this.http.delete(`${baseUrl}/${id}`);
  }

  deleteAll(): Observable<any> {
    return this.http.delete(baseUrl);
  }

  findByName(title: any): Observable<Speaker[]> {
    return this.http.get<Speaker[]>(`${baseUrl}?title=${title}`);
  }
}
